extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -695,16 +790,218 @@ export class AmpLightboxGallery extends AMP.BaseElement {
     }
   }
 
+  /**
+   * Carries momentum for the swipe forwards to a final destination, with the
+   * duration depending on the velocity.
+   * @param {number} scale The current scale.
+   * @param {number} deltaX How far in the x direction we should keep moving.
+   * @param {number} deltaY How far in the y direction we should keep moving.
+   * @param {number} velocity The current velocity.
+   * @return {Promise} A Promise that resolves once the momentum based movement
+   *    based movement has ended.
+   * @private
+   */
+  carrySwipeMomentum_(scale, deltaX, deltaY, velocity) {
+    const duration = velocity * SWIPE_TO_CLOSE_DISTANCE_TO_TIME_FACTOR;
+
+    setStyles(devAssert(this.carousel_), {
+      transform: `scale(${scale}) translate(${deltaX}px, ${deltaY}px)`,
+      transition: `${duration}ms transform ${SWIPE_TO_CLOSE_MOMENTUM_TIMING}`,
+    });
+
+    return delayAfterDeferringToEventLoop(this.win, duration);
+  }
+
+  /**
+   * Snaps back to the starting point, with the duration based on the distance
+   * that needs to be travelled.
+   * @param {number} finalDistance
+   * @return {Promise} A Promise that resolves once the snapping has completed.
+   * @private
+   */
+  snapBackFromSwipe_(finalDistance) {
+    const duration = finalDistance * SWIPE_TO_CLOSE_SNAP_BACK_TIME_FACTOR;
+
+    return this.mutateElement(() => {
+      setStyles(devAssert(this.carousel_), {
+        transform: '',
+        transition: `${duration}ms transform ease-out`,
+      });
+      setStyles(devAssert(this.mask_), {
+        opacity: '',
+        transition: `${duration}ms opacity ease-out`,
+      });
+      setStyles(devAssert(this.controlsContainer_), {
+        opacity: '',
+        transition: `${duration}ms opacity ease-out`,
+      });
+    }).then(() => {
+      return delayAfterDeferringToEventLoop(this.win, duration);
+    }).then(() => {
+      return this.mutateElement(() => {});
```

Comment:
Is this intentional? I think we have a different API for marking an element as dirty if that's what you're looking for.

Reply:
It was need at one point, but it isn't needed anymore, so I just removed it. I really should have commented it though.